enhancements/network/on-prem-service-load-balancers.md

```diff
@@ -0,0 +1,405 @@
+---
+title: on-prem-service-load-balancers
+authors:
+  - "@russellb"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-05-14
+last-updated: 2020-05-14
+status: proposed
+---
+
+# Service Load Balancers for On Premise Infrastructure
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+We do not currently support [Services of
```

Comment:
Just FYI, we have a funny feature around this: "IngressIPs", where we automatically assign ExternalIPs to services of type LoadBalancer if configured. See https://github.com/openshift/api/blob/master/config/v1/types_network.go#L99

Reply:
external IP can be set regardless of service type. For LoadBalancer services you have to set the ingress field in status.
```
status:
  loadBalancer:
    ingress:
    - ip: 203.0.113.1
```

Reply:
Interesting, I didn't know about this feature.  I will incorporate this into the doc.  I imagine this was done before enhancements?  Any other docs I can refer to?

As you implied, I suppose the benefit of this proposal would be a way to make the IPs accessible automatically, either via ARP/NDP or BGP, so it still seems worth pursuing to me.  Would you agree?

Reply:
@russellb, this is [more about ExternalIP on 4.x](https://github.com/openshift/openshift-docs/pull/21388), but I haven't had time to complete it.

Reply:
I've added some notes about this existing functionality and clarified that we lack today is full automation for these services and not that they are completely unusable.  This enhancement adds built-in automation for making these IPs reachable in the right places.